#
# Copyright (c) 2024 Huang Qinjin
#
# Permission to use, copy, modify, and/or distribute this software for any
# purpose with or without fee is hereby granted, provided that the above
# copyright notice and this permission notice appear in all copies.
#
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

[PATCH] Look for the latest Windows SDK in the VS installation root in addition to the registry.

diff --git a/MSBuild/Microsoft/VC/v170/Microsoft.Cpp.WindowsSDK.props b/MSBuild/Microsoft/VC/v170/Microsoft.Cpp.WindowsSDK.props
--- a/MSBuild/Microsoft/VC/v170/Microsoft.Cpp.WindowsSDK.props
+++ b/MSBuild/Microsoft/VC/v170/Microsoft.Cpp.WindowsSDK.props
@@ -119,7 +119,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <SDKVersion Condition="'$(SDKVersion)' == ''">10.0</SDKVersion>
     <SDKDisplayName Condition="'$(SDKDisplayName)' == ''">Windows 10</SDKDisplayName>
 
-    <_LatestWindowsTargetPlatformVersion>$([Microsoft.Build.Utilities.ToolLocationHelper]::GetLatestSDKTargetPlatformVersion($(SDKIdentifier), $(SDKVersion)))</_LatestWindowsTargetPlatformVersion>
+    <_LatestWindowsTargetPlatformVersion>$([Microsoft.Build.Utilities.ToolLocationHelper]::GetLatestSDKTargetPlatformVersion($(SDKIdentifier), $(SDKVersion), $(MsvcWine_WindowsSdkRoots.Split(';', StringSplitOptions.RemoveEmptyEntries))))</_LatestWindowsTargetPlatformVersion>
     <WindowsTargetPlatformVersion Condition="'$(WindowsTargetPlatformVersion)' == '' and '$(_LatestWindowsTargetPlatformVersion)' != ''">10.0</WindowsTargetPlatformVersion>
   </PropertyGroup>
 
